File: src/csharp/Intel/Iced.UnitTests/Intel/FormatterTests/FormatterOptionsTests.cs
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

#if GAS || INTEL || MASM || NASM || FAST_FMT
using System.Collections.Generic;
using System.Linq;

namespace Iced.UnitTests.Intel.FormatterTests {
	static class FormatterOptionsTests {
		public static readonly (OptionsTestCase[] testCases, HashSet<int> ignored) CommonTests = ReadAllTests("Options.Common.txt", "Formatter");
		public static readonly (OptionsTestCase[] testCases, HashSet<int> ignored) AllTests = ReadAllTests("Options.txt", "Formatter");

		static (OptionsTestCase[] testCases, HashSet<int> ignored) ReadAllTests(string filename, params string[] directories) {
			var optionsFilename = PathUtils.GetTestTextFilename(filename, directories);
			var ignored = new HashSet<int>();
			return (OptionsTestsReader.ReadFile(optionsFilename, ignored).ToArray(), ignored);
		}
	}
}
#endif

File: src/java/iced-x86/src/test/java/com/github/icedland/iced/x86/fmt/FormatterOptionsTests.java
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

package com.github.icedland.iced.x86.fmt;

import java.util.HashSet;

import com.github.icedland.iced.x86.PathUtils;

final class FormatterOptionsTests {
	public static class Info {
		public OptionsTestCase[] tests;
		public HashSet<Integer> ignored;
	}

	public static final Info commonTests;
	public static final Info allTests;

	static {
		commonTests = readAllTests("Formatter", "Options.Common.txt");
		allTests = readAllTests("Formatter", "Options.txt");
	}

	private static Info readAllTests(String dir, String filename) {
		String optionsFilename = PathUtils.getTestTextFilename(dir, filename);
		HashSet<Integer> ignored = new HashSet<Integer>();
		Info info = new Info();
		info.tests = OptionsTestsReader.readFile(optionsFilename, ignored).toArray(new OptionsTestCase[0]);
		info.ignored = ignored;
		return info;
	}
}
